internal/onprem/event_extractor.go

```diff
@@ -59,17 +59,15 @@ func (e *EventExtractor) ExtractEvents(tarFilename string) (chan types.EventEnve
 		return eventChannel, err
 	}
 	defer os.RemoveAll(tmpdir)
-
-	e.logger.WithFields(logrus.Fields{
-		"tmpdir": tmpdir,
-		"file":   tarFilename,
-	}).Info("untarring file into tmpdir")
+	logFields := logrus.Fields{
+		"filename": tarFilename,
+		"tmpdir":   tmpdir,
+	}
+	e.logger.WithFields(logFields).Info("untarring file into tmpdir")
 	e.untarFile(tarFilename, tmpdir)
 
-	e.logger.WithFields(logrus.Fields{
-		"directory": tmpdir,
-	}).Info("extracting events from files in directory")
-	e.extractEvents(tmpdir, eventChannel)
+	e.logger.WithFields(logFields).Info("extracting events from files in directory")
+	e.extractEvents(logFields, tmpdir, eventChannel)
```

Comment:
I don't think this having 2 loggers would work well: 1 I would use inherited fields from the calling function (as you suggest), and when I don't need those fields or want to add more, I would need to use the struct's logger. I think this would lead to confusion, wouldn't you agree?